rules/library.bzl

```diff
@@ -639,13 +640,6 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
         import_vfsoverlays.append(import_name + "_vfs")
 
     for vendored_static_library in kwargs.pop("vendored_static_libraries", []):
-        import_name = "%s-%s-library-import" % (name, paths.basename(vendored_static_library))
-        native.objc_import(
-            name = import_name,
-            archives = [vendored_static_library],
-            tags = _MANUAL,
-        )
-        vendored_deps.append(import_name)
         import_headers = native.glob(
```

Comment:
Is it possible to remove this and use `hdrs` instead - it might break folks, but I think it is more of a declarative behavior: there might be middleground strategy as well - just ignoring this value if you set `hdrs`

Reply:
Yep that makes sense, see my comment above. Let me know what you think is best and i can update accordingly. 

IMO id remove the implicit glob we do and let folks update to use `hdrs`. Were still implicitly constructing the `includes` property for `objc_import` but this way we dont make assumptions about where the headers are.